distributed/tests/test_worker.py

```diff
@@ -3235,78 +3235,96 @@ async def test_remove_replicas_simple(c, s, a, b):
     client=True,
     config={"distributed.comm.recent-messages-log-length": 1_000},
 )
-async def test_remove_replicas_while_computing(c, s, *workers):
-    futs = c.map(inc, range(10), workers=[workers[0].address])
-
-    # All interesting things will happen on that worker
-    w = workers[1]
-    intermediate = c.map(slowinc, futs, delay=0.05, workers=[w.address])
-
-    def reduce(*args, **kwargs):
-        import time
+async def test_remove_replicas_while_computing(c, s, a, b):
+    futs = c.map(inc, range(10), workers=[a.address])
+    dependents_event = distributed.Event()
 
-        time.sleep(0.5)
-        return
+    def some_slow(x, event):
+        if x % 2:
```

Comment:
otherwise we might want to implement a `Counter`. I think relying on synchronization primitives for our tests would be a welcome change. So far, whenever I used them this was a great experience (although the different APIs between asyncio.Event and distributed.Event can be consusing since you need to await everything on the distributed API, obviously)

Reply:
x==0 only works if we await that one future before we schedule the others.
Alternatively we can start 6 threads.

Reply:
See if you like it now